crates/vault-registry/src/types.rs

```diff
@@ -457,11 +461,19 @@ impl<T: Config> RichVault<T> {
         })
     }
 
+    pub(crate) fn slash_for_to_be_redeemed(&mut self, amount: &Amount<T>) -> DispatchResult {
+        let vault_id = self.id();
+        let collateral = self.get_vault_collateral()?.min(amount)?;
```

Comment:
There was a bug that caused this (now fixed) but this may also occur if `total_current_stake > vault_collateral` - i.e. all to-be-redeemed are to be refunded included that covered by nomination.